src/optics/CloudOptics.jl

```diff
@@ -201,14 +201,19 @@ end
     build_cloud_mask!(cld_mask, cld_frac, random_arr, gcol, ::MaxRandomOverlap)
 
 Builds McICA-sampled cloud mask from cloud fraction data for maximum-random overlap
+
+Reference: https://github.com/AER-RC/RRTMG_SW/
 """
 function build_cloud_mask!(cld_mask, cld_frac, random_arr, gcol, ::MaxRandomOverlap)
 
     FT = eltype(cld_frac)
-    local_rand = view(random_arr, :, gcol)
+    local_rand = view(random_arr, :, :, gcol)
     local_cld_frac = view(cld_frac, :, gcol)
     local_cld_mask = view(cld_mask, :, :, gcol)
-    ngpt, nlay = length(local_rand), length(local_cld_frac)
+    ngpt, nlay = size(local_rand, 1), length(local_cld_frac)
```

Comment:
```suggestion
    ngpt, nlay = size(local_rand, 1), size(local_rand, 2)
```
Would this be slightly better (easier to understand)? I don't have a preference, up to you.